sources/osgShader/ShaderGenerator.js

```diff
@@ -100,7 +101,7 @@ ShaderGenerator.prototype = {
 
             for (var i = 0; i < cacheType.length; i++) {
                 var type = cacheType[i];
-                var attributeStack = textureUnit[type];
+                var attributeStack = type < textureUnit.length ? textureUnit[type] : undefined;
```

Comment:
(same note about textureUnit.length and _attributeArrayForUnit.length )

Reply:
fixed